p2p/cuprate-p2p/src/config.rs

```diff
@@ -0,0 +1,22 @@
+/// P2P config.
+#[derive(Clone, Debug)]
+pub struct P2PConfig {
+    /// The number of outbound connections to make and try keep.
+    pub outbound_connections: usize,
+    /// The absolute maximum number of held outbound connections.
+    ///
+    /// *Note:* Cuprate might make more connections than this to see if a peer is reachable or
+    /// to get peers from that node, these connections are not held for long though.
+    pub max_outbound_connections: usize,
```

Comment:
Is there anything asserting `max_outbound_connections >= outbound_connections`?

`P2PConfig::allowed_extra_connections` will panic if not.